force-app/main/default/classes/NextStepFuture.cls

```diff
@@ -38,12 +77,15 @@ public with sharing class NextStepFuture {
     * Update the corresponding test class AsynchronousTest and test method nextStepFuture_testCreateDealAndUpdateRole to cover test this method
     */
     //Mix DML Error happens when you try to perform DML operations on setup and non-setup objects in a single transaction.
+    @future
     public static void createDealAndUpdateRole() {
         Opportunity bigDeal = new Opportunity();
         insertBigDeal(bigDeal);
 
         UserRole[] roles = [SELECT Id, Name FROM UserRole WHERE Id = :UserInfo.getUserRoleId() OR Name = 'CRO' LIMIT 1];
-        updateUserRoleName(roles);
+        System.runAs(new User(Id = UserInfo.getUserId())) {
+            updateUserRoleName(roles);
+        }
     }
```

Comment:
This is an interesting, but unorthodox way of solving the problem. The more traditional way would be to decorate `updateUserRole()` as an `@future` method. Both methods have the same result - a new transaction for the code to run in which solves the mixed DML issue. As always, I don't mind unorthodox code, just highlighting the "expectation". But your code works so rock on!